src/hu/elte/txtuml/stdlib/util/regex/MatcherImplementation.java

```diff
@@ -0,0 +1,273 @@
+package RegularExpression;
+
+public class MatcherImplementation implements  Matcher{
+	
+	java.util.regex.Matcher matcher;
+	
+	public MatcherImplementation(java.util.regex.Matcher matcher){
```

Comment:
We will not be able to provide a java.util.regex.Matcher in a model as constructor parameter. The matcher should be created inside this constructor.